src/main/kotlin/fi/metatavu/lipsanen/projects/milestones/tasks/TaskEntity.kt

```diff
@@ -32,6 +33,22 @@ class TaskEntity : Metadata() {
     @Enumerated(EnumType.STRING)
     lateinit var status: TaskStatus
 
+    @OneToMany(mappedBy = "task", cascade = [CascadeType.ALL], orphanRemoval = true)
+    lateinit var assignees: List<TaskAssigneeEntity>
```

Comment:
There's 2 ways to do it:
1 - manage the relation from the children's side, it is often the way we prefer for more clarity but it takes a bit more code. that way means: 
creating TaskAttachmentRepository, TaskAssigneeRepository
adding create, listByTask methods there
removing this "assignees" and "attachmentUrls" fields from this class
keeping @ManyToOne relations in the children classes (@JoinColumn that were used aren't needed)
when creating/editing task, explicitly manage those entities, e.g. create task, save it, create the child entities with references to it.

2 - manage the relation from parent side like you are doing here, so you don't really need any @ManyToOne relations from the child classes and can remove them completely since you never request those fields. Those objects will be saved as long as you create TaskEntity like this 
id = UUID.randomUUID()
assignees = arrayListOf(TaskAssigneeEntity(...))
attachmentUrls = arrayListOf(TaskAttachmentEntity(...))

You can choose whichever you like, by default @ManyToOne should load entities in lazy way, so method 2 does not mean that those extra tables will be queried every time